base_onchange_rule/models/onchange_rule.py

```diff
@@ -0,0 +1,120 @@
+# coding: utf-8
+# © 2017 David BEAL @ Akretion
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from lxml import etree
+from collections import defaultdict
+
+from odoo.osv import orm
+from odoo import api, fields, models
+# from odoo.exceptions import UserError
+
+import logging
+_logger = logging.getLogger(__name__)
+
+
+class OnchangeRule(models.Model):
+    _name = "onchange.rule"
+
+    @api.model
+    def _default_company(self):
+        return self.env['res.company']._company_default_get()
+
+    name = fields.Char(required=True)
+    active = fields.Boolean(
+        default=True,
+        help="The active field allows you to make your record "
+             "without effect but without remove it.")
+    company_id = fields.Many2one(
+        comodel_name='res.company', string='Company', index=1,
+        default=_default_company)
+    model_id = fields.Many2one(
+        comodel_name='ir.model', string='Model', required=True,
+        help="")
+    field_id = fields.Many2one(
+        comodel_name='ir.model.fields', string='Onchange Field',
+        domain="[('model_id', '=', model_id)]", required=True)
+    implied_model = fields.Char(
+        string='Implied Model',
+        related='field_id.relation',
+        help="")
+    readonly = fields.Boolean(
+        help="If checked ensure than record can't be updated by user "
+             "(only by inserted data)")
+    sequence = fields.Integer()
+    line_ids = fields.One2many(
+        comodel_name='onchange.rule.line', inverse_name='onchange_rule_id',
+        string='Onchange Rule Lines')
+
+    @api.onchange('model_id')
+    def _onchange_field(self):
+        self.field_id = False
+
+    @api.model
+    def _customize_view_according_to_setting_rule(
+            self, res, view_type, src_model):
+        if view_type == 'form':
+            rules = self._get_rules_from_model(src_model._name)
+            if rules:
+                doc = etree.XML(res['arch'])
+                params, readonly = (self.env['onchange.rule']
+                                    ._get_config_from_model(
+                                        src_model._name, rules))
+                fields_def = src_model.fields_get(
+                    allfields=None, attributes=None)
+                _logger.debug("  >>>> params, reado %s %s ", params, readonly)
+                for field in params.keys():
+                    for param in params[field]:
+                        self._update_nodes_from_rule(
+                            doc, fields_def, field, 'on_change', 1)
+                for field in readonly:
```

Comment:
Be careful we should not set readonly True but something like "attrs: {'readonly': [('my_trigger_onchange_field', 'in', [3, 4..])]"
Also instead of reparsing the xml maybe we can plug the code for adding/updating the "attrs" here : https://github.com/OCA/OCB/blob/10.0/odoo/addons/base/ir/ir_ui_view.py#L700